qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/utils/QuteSearchUtils.java

```diff
@@ -289,11 +292,45 @@ public static void tryToCollectObjectPartOrParameter(String partName, PartNameMa
 
 	private static boolean tryToCollectObjectPart(String partName, PartNameMatcher matcher, Expression parameterExpr,
 			BiConsumer<Node, Range> collector) {
-		ObjectPart objectPart = parameterExpr.getObjectPart();
-		if (isMatch(objectPart, partName, matcher)) {
-			Range range = QutePositionUtility.createRange(objectPart);
-			collector.accept(objectPart, range);
-			return true;
+		boolean result = false;
+		List<Node> nodes = parameterExpr.getExpressionContent();
+		for (Node nodeExpr : nodes) {
+			result |= tryToCollectObjectPart(partName, matcher, nodeExpr, collector);
+		}

```

Comment:
do we really need to go through all nodes after we found one valid match?

Reply:
I have add comments for this code, hope you will understand more. The code collect all object parts inside nodeExpr. The object part can be declared inside an expression or a parameter method, so we need to loop all nodes.